tests/components/rainforest_raven/test_sensor.py

```diff
@@ -20,3 +28,74 @@ async def test_sensors(
     assert len(hass.states.async_all()) == 5
 
     await snapshot_platform(hass, entity_registry, snapshot, mock_entry.entry_id)
+
+
+@pytest.mark.usefixtures("mock_entry")
+async def test_device_update_error(
+    hass: HomeAssistant,
+    mock_device: AsyncMock,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test handling of a device error during an update."""
+    mock_device.get_network_info.side_effect = (RAVEnConnectionError, NETWORK_INFO)
+
+    states = hass.states.async_all()
+    assert len(states) == 5
+    assert all(state.state != STATE_UNAVAILABLE for state in states)
+
+    freezer.tick(timedelta(seconds=60))
+    await hass.async_block_till_done()
+
+    states = hass.states.async_all()
+    assert len(states) == 5
+    assert all(state.state == STATE_UNAVAILABLE for state in states)
+
+    freezer.tick(timedelta(seconds=60))
+    await hass.async_block_till_done()
```

Comment:
Should we fire a time changed event instead?

https://github.com/home-assistant/core/blob/94506c3c90bd3bd1e289880edf0bf6bd9fbbd900/tests/common.py#L461

I think we normally tick, then fire the time changed event and then block till done (if needed).

Reply:
This PR became a mess as I tried to work around various deadlocks.
I agree this still needs some tweaks, so I've opened a preliminary simpler PR #128652 with just the device info changes.

Reply:
> This PR became a mess as I tried to work around various deadlocks. I agree this still needs some tweaks, so I've opened a preliminary simpler PR #128652 with just the device info changes.

All good now!
Rebased, and coverage is still 100%